cra-rxjs-styled-components/src/hooks/repository-issues/use-repository-issues.ts.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from 'react';
+import { forkJoin, tap } from 'rxjs';
+import { ISSUE_PR_SEARCH } from '../../constants/url.constants';
+import { IssueTypes } from '../../types/types';
+import { fromFetchWithAuth } from '../auth/from-fetch-with-auth';
+
+export function useRepositoryIssues(
+  username: string,
+  repo: string
+): IssueTypes {
+  const [issues, setIssues] = useState<IssueTypes>({
+    closed: {
+      total_count: 0,
+      incomplete_results: true,
+      items: [],
+    },
+    open: {
+      total_count: 0,
+      incomplete_results: true,
+      items: [],
+    },
+  });
+
+  const request = (url: string) =>
+    fromFetchWithAuth(url, {
+      selector: (response: Response) => {
+        return response.json();
+      },
+    });
+
+  useEffect(() => {
+    forkJoin([
+      request(`${ISSUE_PR_SEARCH(username, repo, 'issue', 'open', 20, 1)}`),
```

Comment:
This looks good. Maybe the hook should take `ISSUE_PR_SEARCH` as an argument? So that this hook can also be used to get pull-request data. Right now, it's not so generic.